observer-errors/README.md

```diff
@@ -0,0 +1,130 @@
+
+```js
+new ResizeObserver({}); // TypeError
+```
+
+#### <a id="effects"></a> Effects within the observable elements
+
+Errors due to the resize effect of observable elements and the resize loop (over the boundaries, etc).
+
+Some of these errors may already be recognized by each of Observer API's explainer, like the ResizeObserver's [Deliver Resize Loop error notification](https://drafts.csswg.org/resize-observer-1/#deliver-resize-error).
+
+#### Callback executions
+
+Execution errors when the Observer callback function is called. When unhandled, they bubble out of the callback function execution and eventually hit as an window error.
+
+One way to exemplify this would be an immediate throwing expression:
+
+```js
+new ResizeObserve(function() {
+  throw new Error();
+});
+```
+
+The error from the callback function cannot be captured from registering the elements through the `.observe` method. The registration does not imply a call to that function.
+
+```js
+const observer = new ResizeObserver(function() {
+    throw new Error();
+});
+
+try {
+    observer.observe(elem1);
+} catch (e) {
+    /* the only error to be captured here is if elem1 don't meet the parameters requirements */
+}
+```
+
+The current Observer model has no control over the given error. This error would bubble out and eventually captured in the `window.onerror`. It would be useful for the Observer instances to provide some control to capture and/or handle the error.
+
+While the `.onerror` remains the classic approach for handling errors in window and elements, it doesn't fit well in the Observer the same way it can't add an event listener as in `window.addEventListener`. This matches the idea on [observers API not being based in events](https://github.com/WICG/resize-observer/blob/master/explainer.md#why-an-observer-based-api-and-not-events).
+
+This proposal tries a less complex solution that uses another callback for catching the errors.
+​
+```js
+var observer = new ResizeObserver(function(entries) { /* ... */ });
+observer.observe(elem1);
+observer.observe(elem2);
+observer.observe(elem3);
+
+// The new proposed method
+observer.catch(function(err) {
+    /* error to be captured when the callback is executed */
+});
+```
+
+This `catch` name is heavily inspired on the Promise's API but naming can be discussed and the current choice is not a hard requirement.
+
+It's important to observe that registering the `.catch` callback can remove the bubbling out. It may work as an opt-in to ignore errors:
+
+```js
+observer.catch(function(err) { /* noop */ });
+```
+
+This method captures errors triggered in the callback function. E.g.:
+
+```js
+var observer = new ResizeObserver(function(entries) {
+    throw new Error('meep');
+});
+observer.observe(elem1);
+observer.observe(elem2);
+
+// The new proposed method
+observer.catch(function(err) {
+    console.log(err.message); // 'meep'
+});
+```
```

Comment:
It might be worth calling out here that (AIUI) there is no `*Observer` that will immediately/synchronously invoke its callback. E.g.:

```js
var observer = new ResizeObserver(() => { throw new Error() })
observer.observe(/* ... */) // This cannot possibly invoke the callback
observer.catch(/* ... */)
```

If it _were_ the case that some `*Observer` could, in some circumstance, invoke the callback immediately/synchronously on `.observe()`, then developers would have to be careful to call `.catch()` before `.observe()`. But since that's not the case (as I believe it is for Mutation/Resize/Intersection Observers), there's nothing to worry about here.